lib/DxilCompression/miniz.c

```diff
@@ -1102,7 +1102,9 @@ static mz_bool tdefl_compress_lz_codes(tdefl_compressor *d)
         if (flags & 1)
         {
             mz_uint s0, s1, n0, n1, sym, num_extra_bits;
-            mz_uint match_len = pLZ_codes[0], match_dist = *(const mz_uint16 *)(pLZ_codes + 1);
+            mz_uint16 read_val;
+            memcpy((void*)&read_val, pLZ_codes + 1, sizeof(mz_uint16));
+            mz_uint match_len = pLZ_codes[0], match_dist = read_val;
```

Comment:
Unsure why this change is necessary.  Is the issue that `pLZ_codes + 1` (plus 2 bytes read from there) is potentially out of buffer bounds at this point (according to UBSan's knowledge)?  Does the memcpy simply obfuscate things so UBSan doesn't complain?

Reply:
Not potentially... UBSan only complains on actual occurred runtime errors.

By definition memcpy will perform byte-by-byte copy, it _may_ optimize for aligned data reads but is conservative.